File: apps/ios/ExpCore/Sources/API/OnboardingApi.swift
import Foundation

private struct EmptyInput: Encodable {}

public final class OnboardingApi: Sendable {
    private let trpc: TrpcClient

    public init(trpc: TrpcClient) {
        self.trpc = trpc
    }

    // Marks onboarding complete for the current user (onboarding.complete in the
    // shared appRouter — sets users.onboardingCompletedAt = now). No input.
    public func complete(accountId: String) async throws {
        try await trpc.mutationVoid(accountId: accountId, path: "onboarding.complete", input: EmptyInput())
    }
}

File: apps/android/app/src/main/java/com/exponential/app/data/api/OnboardingApi.kt
package com.exponential.app.data.api

import javax.inject.Inject
import javax.inject.Singleton
import kotlinx.serialization.Serializable

@Serializable
private object OnboardingEmptyInput

@Singleton
class OnboardingApi @Inject constructor(private val trpc: TrpcClient) {
    // Marks onboarding complete for the current user (onboarding.complete in the
    // shared appRouter — sets users.onboardingCompletedAt = now). No input.
    suspend fun complete(accountId: String) {
        trpc.mutationUnit(
            accountId,
            path = "onboarding.complete",
            input = OnboardingEmptyInput,
            inputSerializer = OnboardingEmptyInput.serializer(),
        )
    }
}
